catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsLoader.java

```diff
@@ -100,32 +103,43 @@ public static boolean enabled() {
    * <p>Caution should be exercised when executing code as system. Results from querying as system
    * are never returned to the client or cached. This means there is no risk of data leak.
    *
-   * @param framework the catalog framework, for creating system templates.
-   * @param util for querying the catalog.
+   * @param catalogFramework the catalog framework, for creating system templates.
+   * @param endpointUtil for querying the catalog.
    * @param systemTemplates system templates loaded from config.
    */
   public static void bootstrap(
-      CatalogFramework framework, EndpointUtil util, List<Metacard> systemTemplates) {
-    Set<String> queryTitles =
-        executeAsSystem(() -> titlesTransform(util.getMetacardsByFilter(QUERY_TEMPLATE_TAG)));
-    Set<String> resultTitles =
-        executeAsSystem(() -> titlesTransform(util.getMetacardsByFilter(ATTRIBUTE_GROUP_TAG)));
+      CatalogFramework catalogFramework,
+      EndpointUtil endpointUtil,
+      List<Metacard> systemTemplates) {
+    Subject systemSubject = SECURITY.runAsAdmin(SECURITY::getSystemSubject);
+    if (systemSubject == null) {
+      throw new SecurityException("Could not get systemSubject to initialize system templates");
+    }
 
-    List<Metacard> dedupedTemplateMetacards =
-        Stream.concat(
-                systemTemplates
-                    .stream()
-                    .filter(QueryTemplateMetacard::isQueryTemplateMetacard)
-                    .filter(metacard -> !queryTitles.contains(metacard.getTitle())),
-                systemTemplates
-                    .stream()
-                    .filter(AttributeGroupMetacard::isAttributeGroupMetacard)
-                    .filter(metacard -> !resultTitles.contains(metacard.getTitle())))
-            .collect(Collectors.toList());
+    ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
 
-    if (!dedupedTemplateMetacards.isEmpty()) {
-      executeAsSystem(() -> framework.create(new CreateRequestImpl(dedupedTemplateMetacards)));
-    }
+    Failsafe.with(
+            new RetryPolicy()
+                .retryOn(Collections.singletonList(RuntimeException.class))
+                .withMaxRetries(5)
+                .withBackoff(2, 60, TimeUnit.SECONDS))
+        .with(executor)
+        .onComplete(
+            (connection, failure) -> {
+              if (connection != null) {
+                LOGGER.info("Successfully loaded System Search Forms", connection);
```

Comment:
❓Why is this called `connection`? And what is expected to be passed to us here since we're executing a `Runnable` that doesn't return a result?

Reply:
Also where is the parameter to this logging statement going? Do you need a set of curly braces?

Reply:
renamed and added parameter to logging.  the rest of this behavior is taken from the failsafe api.  i believe it's a CheckedBiConsumer http://jodah.net/failsafe/javadoc/net/jodah/failsafe/function/CheckedBiConsumer.html